versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -1304,4 +1449,168 @@ protected boolean metadataUpperCase() {
   protected boolean batchDDL() {
     return false;
   }
+
+  protected Hash writeRefLogEntry(
+      Connection connection,
+      String refName,
+      AdapterTypes.RefLogEntry.RefType refType,
+      Hash parentRefLogId,
+      Hash commitHash,
+      AdapterTypes.RefLogEntry.Operation operation,
+      long timeInMicros,
+      List<Hash> sourceHashes)
+      throws ReferenceConflictException {
+
+    Hash currentRefLogId = randomHash();
+    RefLog parentEntry = fetchFromRefLog(connection, parentRefLogId);
```

Comment:
It looks like you can avoid fetching the parelnt-ref-log-id in all call-sites of this function, because `fetchFromRefLog` handles the "null-case"?